active/0002-centalized-mgmt-resource-types-provider-versions.md

```diff
@@ -274,4 +251,8 @@ At this time we are not proposing support for wildcard semantics on provider ver
 
 ### Deciding if `sys` is a compile time import or a resource types provider package
 
-It is possible to think of `sys` more like a compile-time namespace import rather than a provider, since all it contains are some built-in functions. We used to share the keyword import between providers and imports, but now that we changed the keyword for providers, at some point we will need to update the implementation for sys to make it a compile-time import. This is out of scope of this REP however.
\ No newline at end of file
+It is possible to think of `sys` more like a compile-time namespace import rather than a provider, since all it contains are some built-in functions. We used to share the keyword import between providers and imports, but now that we changed the keyword for providers, at some point we will need to update the implementation for sys to make it a compile-time import. This is out of scope of this REP however.
```

Comment:
I don't follow - why is `sys` more like a compile-time import than a provider? My impression is the complete opposite.

Compile-time imports require referencing variables or functions from another Bicep/JSON file. Providers allow referencing behavior from the runtime.

Reply:
As discussed today, I'm good with special casing "sys", and blocking it in the bicepconfig. If we're considering it to not be a provider, should we also block the following?

```bicep
provider sys
```

Reply:
I think since the conversation is still not settled, lets just treat it as any other provider and call out in the docs what are the consequences of doing this so users dont end up shooting themselves in the foot

Reply:
> I think since the conversation is still not settled, lets just treat it as any other provider and call out in the docs what are the consequences of doing this so users dont end up shooting themselves in the foot

I thought we settled this in the discussion today? Here's my recollection:
1. We do not consider "sys" to be a provider - it is instead an intrinsic feature of the runtime. So regular "provider" semantics don't need to apply to it, and we expect this to fit with the user's mental model of what a provider is.
1. To prevent weird behavior, we should block "sys" in the "providers" configuration section introduced by this rep.
1. "sys" should not appear or be required in the "implicitProviders" config we discussed.

We didn't discuss this today, but to me the following is also necessary to be consistent with the above:
1. We should block usage of the syntax `provider sys`

If we're happy with all of the above, then I think we can document it in this rep and move on.